ports/stm32/boards/ARDUINO_OPTA/mpconfigboard.h

```diff
@@ -0,0 +1,239 @@
+#define MICROPY_HW_RCC_I2C123_CLKSOURCE (RCC_I2C123CLKSOURCE_D2PCLK1)
+#define MICROPY_HW_RCC_QSPI_CLKSOURCE   (RCC_QSPICLKSOURCE_PLL2)
+
+// SMPS configuration
+#define MICROPY_HW_PWR_SMPS_CONFIG      (PWR_LDO_SUPPLY)
+
+// Configure the analog switches for dual-pad pins.
+#define MICROPY_HW_ANALOG_SWITCH_PA0    (SYSCFG_SWITCH_PA0_OPEN)
+#define MICROPY_HW_ANALOG_SWITCH_PA1    (SYSCFG_SWITCH_PA1_OPEN)
+#define MICROPY_HW_ANALOG_SWITCH_PC2    (SYSCFG_SWITCH_PC2_OPEN)
+#define MICROPY_HW_ANALOG_SWITCH_PC3    (SYSCFG_SWITCH_PC3_OPEN)
+
+// There is an external 32kHz oscillator
+#define RTC_ASYNCH_PREDIV           (0)
+#define RTC_SYNCH_PREDIV            (0x7fff)
+#define MICROPY_HW_RTC_USE_BYPASS   (1)
+#define MICROPY_HW_RTC_USE_US       (1)
+#define MICROPY_HW_RTC_USE_CALOUT   (1)
+
+#if !MICROPY_HW_ENABLE_INTERNAL_FLASH_STORAGE
+// QSPI flash #1 for storage
+#define MICROPY_HW_QSPI_PRESCALER           (2) // 100MHz
+#define MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2 (27)
+// Reserve 1MiB at the end for compatibility with alternate firmware that places WiFi blob here.
+#define MICROPY_HW_SPIFLASH_SIZE_BITS       (120 * 1024 * 1024)
+#define MICROPY_HW_QSPIFLASH_CS             (pyb_pin_QSPI2_CS)
+#define MICROPY_HW_QSPIFLASH_SCK            (pyb_pin_QSPI2_CLK)
+#define MICROPY_HW_QSPIFLASH_IO0            (pyb_pin_QSPI2_D0)
+#define MICROPY_HW_QSPIFLASH_IO1            (pyb_pin_QSPI2_D1)
+#define MICROPY_HW_QSPIFLASH_IO2            (pyb_pin_QSPI2_D2)
+#define MICROPY_HW_QSPIFLASH_IO3            (pyb_pin_QSPI2_D3)
+
+// SPI flash #1, block device config
+extern const struct _mp_spiflash_config_t spiflash_config;
+extern struct _spi_bdev_t spi_bdev;
+#define MICROPY_HW_BDEV_SPIFLASH    (&spi_bdev)
+#define MICROPY_HW_BDEV_SPIFLASH_CONFIG (&spiflash_config)
+#define MICROPY_HW_BDEV_SPIFLASH_SIZE_BYTES (MICROPY_HW_SPIFLASH_SIZE_BITS / 8)
+#define MICROPY_HW_BDEV_SPIFLASH_EXTENDED (&spi_bdev)
+#endif
+
+// 4 wait states
+#define MICROPY_HW_FLASH_LATENCY    FLASH_LATENCY_2
+
+// UART config
+#define MICROPY_HW_UART3_TX         (pyb_pin_UART3_TX)
+#define MICROPY_HW_UART3_RX         (pyb_pin_UART3_RX)
+#define MICROPY_HW_UART3_RTS        (pyb_pin_UART3_RTS)
+#define MICROPY_HW_UART3_CTS        (pyb_pin_UART3_CTS)
+
+#define MICROPY_HW_UART4_TX         (pyb_pin_BT_TXD)
+#define MICROPY_HW_UART4_RX         (pyb_pin_BT_RXD)
+#define MICROPY_HW_UART4_RTS        (pyb_pin_BT_RTS)
+#define MICROPY_HW_UART4_CTS        (pyb_pin_BT_CTS)
+
+#define MICROPY_HW_UART6_TX         (pyb_pin_UART6_TX)
+#define MICROPY_HW_UART6_RX         (pyb_pin_UART6_RX)
+
+// I2C buses
+#define MICROPY_HW_I2C1_SCL         (pyb_pin_I2C1_SCL)
+#define MICROPY_HW_I2C1_SDA         (pyb_pin_I2C1_SDA)
+
+#define MICROPY_HW_I2C3_SCL         (pyb_pin_I2C3_SCL)
+#define MICROPY_HW_I2C3_SDA         (pyb_pin_I2C3_SDA)
+
+// USRSW is pulled low. Pressing the button makes the input go high.
+#define MICROPY_HW_USRSW_PIN        (pyb_pin_USER_BUTTON)
```

Comment:
The comment here doesn't match the settings, which seem to indicate it's pulled high and pressing makes it go low.